components/org.wso2.carbon.identity.api.server.asynchronous.operation.status.management/org.wso2.carbon.identity.api.server.asynchronous.operation.status.management.v1/src/main/java/org/wso2/carbon/identity/api/server/asynchronous/operation/status/management/v1/core/AsyncOperationsApiServiceCore.java

```diff
@@ -0,0 +1,329 @@
+import static org.wso2.carbon.identity.api.server.asynchronous.operation.status.management.v1.constants.AsyncOperationStatusMgtEndpointConstants.FILTER_PARAM;
+import static org.wso2.carbon.identity.api.server.asynchronous.operation.status.management.v1.constants.AsyncOperationStatusMgtEndpointConstants.LIMIT_PARAM;
+import static org.wso2.carbon.identity.api.server.asynchronous.operation.status.management.v1.constants.AsyncOperationStatusMgtEndpointConstants.NEXT;
+import static org.wso2.carbon.identity.api.server.asynchronous.operation.status.management.v1.constants.AsyncOperationStatusMgtEndpointConstants.PAGINATION_AFTER;
+import static org.wso2.carbon.identity.api.server.asynchronous.operation.status.management.v1.constants.AsyncOperationStatusMgtEndpointConstants.PAGINATION_BEFORE;
+import static org.wso2.carbon.identity.api.server.asynchronous.operation.status.management.v1.constants.AsyncOperationStatusMgtEndpointConstants.PATH_SEPARATOR;
+import static org.wso2.carbon.identity.api.server.asynchronous.operation.status.management.v1.constants.AsyncOperationStatusMgtEndpointConstants.PREVIOUS;
+import static org.wso2.carbon.identity.api.server.asynchronous.operation.status.management.v1.util.AsyncOperationStatusEndpointUtil.buildAsyncStatusMgtClientException;
+import static org.wso2.carbon.identity.api.server.asynchronous.operation.status.management.v1.util.AsyncOperationStatusEndpointUtil.buildURIForPagination;
+
+/**
+ * Core service class for handling asynchronous operation status management APIs.
+ */
+public class AsyncOperationsApiServiceCore {
+
+    private final AsyncOperationStatusMgtService asyncOperationStatusMgtService;
+
+    private static final Log LOG = LogFactory.getLog(AsyncOperationsApiServiceCore.class);
+
+    public AsyncOperationsApiServiceCore(AsyncOperationStatusMgtService asyncOperationStatusMgtService) {
+
+        this.asyncOperationStatusMgtService = asyncOperationStatusMgtService;
+    }
+
+    public Response getOperations(String after, String before, Integer limit, String filter) {
+
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            limit = validateLimit(limit);
+            List<OperationResponseDTO> records =
+                    asyncOperationStatusMgtService.getOperations(tenantDomain, after, before, limit + 1, filter);
+            return Response.ok().entity(getOperationsResponse(limit, after, before, filter, records)).build();
+        } catch (AsyncOperationStatusMgtException e) {
+            throw AsyncOperationStatusEndpointUtil.handleAsyncOperationStatusMgtException(e);
+        }
+    }
+
+    public Response getOperation(String operationId) {
+
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        if (StringUtils.isBlank(operationId)) {
+            return Response.status(Response.Status.BAD_REQUEST).build();
```

Comment:
Handle as a client exception by defining a Error constant. Needs to communicate the error details.